pageserver/src/tenant/size.rs

```diff
@@ -40,6 +41,26 @@ struct TimelineInputs {
     next_gc_cutoff: Lsn,
 }
 
+/// Gathers the inputs for the tenant sizing model.
+///
+/// Tenant size does not consider the latest state, but only the state until next_gc_cutoff, which
+/// is updated on-demand, during the start of this calculation and separate from the
+/// [`Timeline::latest_gc_cutoff`].
+///
+/// For branches starting from zero, having no ancestor_timeline:
+/// ```ignore
+/// 0-----|---------|----|------------| · · · · · |·> lsn
+///   initdb_lsn  branchpoints*  next_gc_cutoff  latest
+/// ```
+///
+/// Until `next_gc_cutoff > initdb_lsn`, tenant's size is 0.
+///
+/// For branched off timelines only such situations are considered:
+///
+/// ```ignore
+///        x----------| · · · · · ·|·> lsn
+/// branchpoint  next_gc_cutoff  latest
+/// ```
```

Comment:
This is confusing as it would need branchpoints as well, removing it in next push.